x-pack/platform/plugins/shared/alerting/server/routes/rule/apis/snooze/external/snooze_rule_route.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import Boom from '@hapi/boom';
+import { v4 } from 'uuid';
+import { IRouter } from '@kbn/core/server';
+import {
+  type SnoozeParams,
+  type SnoozeResponse,
+  snoozeBodySchema,
+  snoozeParamsSchema,
+  snoozeResponseSchema,
+} from '../../../../../../common/routes/rule/apis/snooze';
+import { ILicenseState, RuleMutedError } from '../../../../../lib';
+import { verifyAccessAndContext } from '../../../../lib';
+import { AlertingRequestHandlerContext, BASE_ALERTING_API_PATH } from '../../../../../types';
+import { DEFAULT_ALERTING_ROUTE_SECURITY } from '../../../../constants';
+import {
+  transformCustomScheduleToRRule,
+  transformRRuleToCustomSchedule,
+} from '../../../../../../common/routes/schedule';
+import type { SnoozeRuleOptions } from '../../../../../application/rule/methods/snooze';
+
+export const snoozeRuleRoute = (
+  router: IRouter<AlertingRequestHandlerContext>,
+  licenseState: ILicenseState
+) => {
+  router.post(
+    {
+      path: `${BASE_ALERTING_API_PATH}/rule/{id}/snooze_schedule`,
+      security: DEFAULT_ALERTING_ROUTE_SECURITY,
+      options: {
+        access: 'public',
+        summary: 'Schedule a snooze for the rule',
+        description: 'When you snooze a rule, the rule checks continue to run but alerts will not generate actions. You can snooze for a specified period of time or indefinitely and schedule single or recurring downtimes.',
+        tags: ['oas-tag:alerting'],
+        availability: {
```

Comment:
>        availability: {

I added the `since` and `stability` values in 16ee5a5f68ede6500b1eb70ffc3cd035324de4f6 but am just calling it out in case it ought to be `experimental` instead of `stable`, for example.